front/pages/poke/[wId]/upgrade_enterprise/index.tsx

```diff
@@ -0,0 +1,303 @@
+import { Spinner } from "@dust-tt/sparkle";
+import type {
+  NewEnterpriseSubscriptionType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { NewEnterpriseSubscriptionSchema, removeNulls } from "@dust-tt/types";
+import { ioTsResolver } from "@hookform/resolvers/io-ts";
+import { useRouter } from "next/router";
+import React from "react";
+import type { Control } from "react-hook-form";
+import { useForm } from "react-hook-form";
+
+import PokeNavbar from "@app/components/poke/PokeNavbar";
+import { PokeButton } from "@app/components/poke/shadcn/ui/button";
+import { PokeCheckbox } from "@app/components/poke/shadcn/ui/checkbox";
+import {
+  PokeForm,
+  PokeFormControl,
+  PokeFormField,
+  PokeFormItem,
+  PokeFormLabel,
+  PokeFormMessage,
+} from "@app/components/poke/shadcn/ui/form";
+import { PokeInput } from "@app/components/poke/shadcn/ui/input";
+import { withSuperUserAuthRequirements } from "@app/lib/iam/session";
+
+export const getServerSideProps = withSuperUserAuthRequirements<{
+  owner: WorkspaceType;
+}>(async (context, auth) => {
+  const owner = auth.workspace();
+  if (!owner) {
+    return {
+      notFound: true,
+    };
+  }
+
+  return {
+    props: {
+      owner,
+    },
+  };
+});
+
+export default function NewEnterpriseSubscriptionForm({
+  owner,
+}: {
+  owner: WorkspaceType;
+}) {
+  const router = useRouter();
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [error, setError] = React.useState<string | null>(null);
+
+  const form = useForm<NewEnterpriseSubscriptionType>({
+    resolver: ioTsResolver(NewEnterpriseSubscriptionSchema),
+    defaultValues: {
+      stripeSubscriptionId: "",
+      code: "",
+      name: "",
+      isSlackbotAllowed: true,
+      isSlackAllowed: true,
+      isNotionAllowed: true,
+      isGoogleDriveAllowed: true,
+      isGithubAllowed: true,
+      isIntercomAllowed: true,
+      isConfluenceAllowed: true,
+      isWebCrawlerAllowed: true,
+      maxMessages: -1,
+      dataSourcesCount: -1,
+      dataSourcesDocumentsCount: -1,
+      dataSourcesDocumentsSizeMb: 2,
+      maxUsers: 1000,
+    },
+  });
+
+  const onSubmit = (values: NewEnterpriseSubscriptionType) => {
+    const cleanedValues = Object.fromEntries(
+      removeNulls(
+        Object.entries(values).map(([key, value]) => {
+          if (typeof value !== "string") {
+            return [key, value];
+          }
+          const cleanedValue = value.trim();
+          if (!cleanedValue) {
+            return null;
+          }
+          return [key, cleanedValue];
+        })
+      )
+    );
+
+    void submit();
+    async function submit() {
```

Comment:
For inline definition of function we should assign to const, it makes it easier to read and collapse the code.